scripts/lint/check-jaeger-idl-version.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# Copyright (c) 2023 The Jaeger Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+set -euo pipefail
+
+dependency="github.com/jaegertracing/jaeger-idl"
+version_regex="v[0-9]\.[0-9]\+\.[0-9]\+"
+
+get_gomod_version() {
+  gomod_dep=$(grep $dependency <go.mod)
+  if [ ! "$gomod_dep" ]; then
+    printf "Error: jaeger-idl dependency not found in go mod\n" >&2
+    exit 1
+  fi
+  gomod_version=$(echo "$gomod_dep" | awk '{print $2}')
+  echo "$gomod_version"
+}
+
+get_submodule_version() {
+  cd idl
+  commit_version=$(git rev-parse HEAD)
+  tags=$(git describe --tags --exact-match "$commit_version")
+  if [ ! "$tags" ]; then
+    printf "Error: failed getting version from submodule\n" >&2
+    exit 1
+  fi
+  semver=$(echo "$tags" | grep "$version_regex")
```

Comment:
Why does this matter if we just compare it with go.mod?

Reply:
If the tag matches what's in go.mod then it doesn't matter what the tag value is.